server/schedule/checker/merge_checker.go

```diff
@@ -145,6 +145,13 @@ func (m *MergeChecker) Check(region *core.RegionInfo) []*operator.Operator {
 		return nil
 	}
 
+	// Check whether the store of the target region leader is normal.
+	store := m.cluster.GetStore(target.GetLeader().GetStoreId())
```

Comment:
why only consider leader instead of all peers?

Reply:
After merging, the target Region will extend its range and provide the service, so I think we just need to make sure the leader of the target Region is in a normal store?

Reply:
But the original issue did not mention anything about the leader. It looks the main issue is moving peer to offline store may slow down the offline process.